viaf.py

```diff
@@ -2,60 +2,271 @@
 from nose.tools import set_trace
 from lxml import etree
 import logging
+from fuzzywuzzy import fuzz
 import re
 
 from collections import Counter, defaultdict
 
+from core.metadata_layer import (
+    ContributorData,
+    Metadata, 
+)
+
 from core.model import (
     Contributor,
     DataSource,
     Representation,
 )
 
+from core.util.personal_names import (
+    display_name_to_sort_name, 
+    is_corporate_name, 
+    normalize_contributor_name_for_matching, 
+)
+
+from core.util.titles import (
+    normalize_title_for_matching, 
+    unfluff_title, 
+)
+
 from core.util.xmlparser import (
     XMLParser,
 )
-from core.util.personal_names import display_name_to_sort_name
+
+
+
 
 class VIAFParser(XMLParser):
 
     NAMESPACES = {'ns2' : "http://viaf.org/viaf/terms#"}
 
     log = logging.getLogger("VIAF Parser")
+    log.setLevel(logging.INFO)  # TODO: remove
+    wikidata_id = re.compile("^Q[0-9]")
+
+
+
+    @classmethod
+    def combine_nameparts(self, given, family, extra):
+        """Turn a (given name, family name, extra) 3-tuple into a
+        display name.
+        """
+        if not given and not family:
+            return None
+        if family and not given:
+            display_name = family
+        elif given and not family:
+            display_name = given
+        else:
+            display_name = given + ' ' + family
+        if extra and not extra.startswith('pseud'):
```

Comment:
(Graf means Count, dude was one of those snooty aristocrats who slummed it in the villages and thought themselves spiritual and just.  Not that I'm bitter.)